docs/guide/cli/index.md

```diff
@@ -0,0 +1,253 @@
+# CLI
+
+The e18e CLI is a powerful tool for analyzing and optimizing your JavaScript/TypeScript projects. It helps you identify performance issues, find optimization opportunities, and migrate to better alternatives.
+
+## Quick Start
+
+```bash
+# Install and run analysis
+npx @e18e/cli analyze
+
+# Migrate packages interactively
+npx @e18e/cli migrate --interactive
+```
+
+## Installation
+
+You can install the e18e CLI globally or use it with npx:
+
+::: code-group
+
+```sh [npm]
+npm install -g @e18e/cli
+```
+
+```sh [yarn]
+yarn global add @e18e/cli
+```
+
+```sh [pnpm]
+pnpm add -g @e18e/cli
+```
+
+```sh [bun]
+bun add -g @e18e/cli
+```
+
+:::
+
+## Commands
+
+### `analyze`
+
+Analyzes your project for various issues and provides recommendations for optimization.
+
+```bash
+e18e-cli analyze [path] [options]
+```
+
+**Arguments:**
+- `path` (optional) - Path to a directory or tarball file to analyze. Defaults to current directory.
+
+**Options:**
+- `--pack <type>` - Package manager to use for packing. Options: `auto`, `npm`, `yarn`, `pnpm`, `bun`, `none`. Default: `auto`
+- `--log-level <level>` - Set the log level. Options: `debug`, `info`, `warn`, `error`. Default: `info`
+- `--manifest <path>` - Path(s) to custom manifest file(s) for module replacements analysis
+
+**Examples:**
+```sh
+# Analyze current project
+e18e-cli analyze
+
+# Analyze with debug output
+e18e-cli analyze --log-level debug
+
+# Analyze with specific package manager
+e18e-cli analyze --pack pnpm
+```
+
+### `migrate`
+
+Migrates your project from packages to more performant alternatives.
+
+```bash
+e18e-cli migrate [packages...] [options]
+```
+
+**Arguments:**
+- `packages...` - Names of packages to migrate (e.g., `chalk`, `lodash`)
+
+**Options:**
+- `--all` - Run all available migrations
+- `--dry-run` - Don't apply any fixes, only show what would change
+- `--include <pattern>` - Files to migrate. Default: `**/*.{ts,js}`
+- `--interactive` - Run in interactive mode
+- `--manifest <path>` - Use custom migration manifest file(s)
+
+**Examples:**
+```bash
+# Migrate specific packages
+e18e-cli migrate chalk lodash
+
+# Preview changes without applying
+e18e-cli migrate chalk --dry-run
+
+# Migrate all available packages
+e18e-cli migrate --all
+
+# Interactive mode to select packages
+e18e-cli migrate --interactive
+
+# Use custom migration manifest
+e18e-cli migrate --manifest ./custom-migrations.json
+```
+
+## Migration System
+
+The CLI uses an [opinionated codemod](https://github.com/es-tooling/module-replacements-codemods) approach for migrations, based on e18e's curated list of performance-focused package replacements. This ensures consistent, well-tested migrations across the ecosystem.
+
+### Default Migrations
+The CLI comes with pre-configured migrations for common performance optimizations:
+- `chalk` → `picocolors` (smaller, faster)
```

Comment:
we probably can't maintain a list of available migrations, as it'll grow to be pretty big

i think we can just link to the replacements doc (which currently shows an `x` for all replacements, but we need to update that)